lms/static/js/edxnotes/utils/notes_collector.js

```diff
@@ -0,0 +1,111 @@
+(function(define) {
+    'use strict';
+    define(['jquery', 'underscore', 'annotator_1.2.9'], function($, _, Annotator) {
+        var fetchOnce = null,
+            fetchNotes,
+            renderNotes,
+            _requestsData = null,
+            storeNotesRequestData,
+            SearchRequestsData = [],
+            waitUntilAllRequestsStored;
+
+        waitUntilAllRequestsStored = function (maxTimeout) {
+            var deferred = $.Deferred(),
+                intervalId = null,
+                elapsedTimeout = 0,
+                pollInterval = 100,
+                checkRequests;
+
+            checkRequests = function() {
+               if ($('.edx-notes-wrapper').length === SearchRequestsData.length) {
+                   deferred.resolve();
+               } else {
+                   elapsedTimeout += pollInterval;
+                   if (elapsedTimeout >= maxTimeout) {
+                       clearInterval(intervalId);
+                       deferred.reject();
+                   }
+               }
+            };
+
+            intervalId = setInterval(checkRequests, pollInterval);
+            return deferred.promise();
+        };
+
+        storeNotesRequestData = function (data) {
+            SearchRequestsData.push(data);
+            
+            // We only need to call `fetchNotes` once only.
+            if (SearchRequestsData.length === 1) {
+                fetchNotes();                
+            }
+        };
+
+        fetchNotes = function () {
+            console.log('fetchNotes called');
+            waitUntilAllRequestsStored(5000).done(function () {
```

Comment:
IMO this code should do something if the timeout is reached. Maybe it is okay not to show a message if the idea is that the notes are not critical, but at the least it should be logged.

I'd also make the 5000 into a constant so that it can be controlled.